File: src - java/Esercizi/PushIf/Pila.java
public Pila(int _maxPush) {
	maxPush = _maxPush;
}

private void _push(int el) {
	Record p = new Record();
	p.elem = el;
	p.succ = top;
	top = p;
	numEl++;
}

public boolean push(int el) {
	if (currPush < maxPush) {
		_push(el);
		currPush++;
		return true;
	}
	return false;
}

private int _pop() throws IllegalStateException {
	if (top != null) {
		int el = top.elem; // memorizza il valore di testa
		                   // per restituirlo alla fine
		top = top.succ;
		numEl--;
		return el;
	}
	
	throw new IllegalStateException();
}

public int pop() {
	currPush = 0;
	return _pop();
}

private void _svuota() {
	top = null;
	numEl = 0;
}

public void svuota() {
	currPush = 0;
	_svuota();
}

File: src - c++/Esercizi/PushIf/Pila.cpp
Pila::Pila(unsigned int maxpush):
  top(0), nelem(0), _maxpush(maxpush), _currpush(0) {
}

void Pila::_Push(const TElem& e) {
  //Classica Push() in una pila: metodo privato
  PRec p = new Record;
  p->el = e;
  p->succ = top;
  top = p;
  nelem++;
}  
  
bool Pila::Push(const TElem& e) {
  if (_currpush < _maxpush) {
    _Push(e); //Inserisce incondizionatamente nella pila
    _currpush++;
    return true;
  }  
  
  return false;
}

TElem Pila::Pop() {
  if (top) {
    //memorizza il valore di testa per restituirlo alla fine
    TElem e = top->el;

    //memorizza il puntatore alla testa: essa dovrà essere cancellata
    PRec p = top;
    top = top->succ; //porta la testa al successivo
    delete p; //elimina la vecchia testa
    
    nelem--;   
    _currpush = 0; //azzero il conteggio degli inserimenti
    return e;
  }
  
  //questo metodo restituisce un valore non
  //specificato nel caso la pila sia vuota
}

void Pila::Svuota() {
  while (top) {
    PRec p = top;
    top = top->succ;
    delete p;
  }
  nelem = 0;
  _currpush = 0; //azzero il conteggio degli inserimenti
}
